src/barfturn.ts

```diff
@@ -0,0 +1,270 @@
+        isEmbezzler &&
+        !get("_envyfishEggUsed") &&
+        myLevel() >= 11 &&
+        (getModifier("Adventure Underwater") ||
+          waterBreathingEquipment.some((item) => have(item))) &&
+        (getModifier("Underwater Familiar") ||
+          familiarWaterBreathingEquipment.some((item) => have(item))) &&
+        (have($effect`Fishy`) || (have($item`fishy pipe`) && !get("_fishyPipeUsed"))) &&
+        !have($item`envyfish egg`) &&
+        canAdventure($location`The Briny Deeps`);
+      const targetLocation = shouldGoUnderwater
+        ? $location`The Briny Deeps`
+        : determineDraggableZoneAndEnsureAccess();
+
+      isEmbezzler ? embezzlerPrep() : freeFightPrep();
+      adventureMacroAuto(targetLocation, Macro.basicCombat());
+      return get("_sourceTerminalDigitizeMonsterCount") !== start;
+    },
+  },
+  {
+    name: "Guaranteed Kramco",
+    available: () => sober() && kramcoGuaranteed(),
+    perform: () => {
+      freeFightPrep(new Requirement([], { forceEquip: $items`Kramco Sausage-o-Matic™` }));
+      adventureMacroAuto(determineDraggableZoneAndEnsureAccess(), Macro.basicCombat());
+      return !kramcoGuaranteed();
+    },
+  },
+  {
+    name: "Void Monster",
+    available: () =>
+      sober() &&
+      have($item`cursed magnifying glass`) &&
+      get("cursedMagnifyingGlassCount") === 13 &&
+      get("_voidFreeFights") < 5,
+    perform: () => {
+      freeFightPrep(new Requirement([], { forceEquip: $items`cursed magnifying glass` }));
+      adventureMacroAuto(determineDraggableZoneAndEnsureAccess(), Macro.basicCombat());
+      return get("cursedMagnifyingGlassCount") === 0;
+    },
+  },
+  {
+    name: "Envyfish Egg",
+    available: () =>
+      have($item`envyfish egg`) && get("envyfishMonster") === embezzler && !get("_envyfishEggUsed"),
+    perform: () => {
+      embezzlerPrep();
+      withMacro(Macro.meatKill(), () => use($item`envyfish egg`), true);
+      return get("_envyfishEggUsed");
+    },
+  },
+  {
+    name: "Map for Pills",
+    available: () =>
+      !sober() &&
+      globalOptions.ascending &&
+      clamp(myAdventures() - digitizedMonstersRemaining(), 1, myAdventures()) <=
+        availableAmount($item`Map to Safety Shelter Grimace Prime`),
+    perform: () => {
+      const choiceToSet =
+        availableAmount($item`distention pill`) <
+        availableAmount($item`synthetic dog hair pill`) +
+          availableAmount($item`Map to Safety Shelter Grimace Prime`)
+          ? 1
+          : 2;
+      setChoice(536, choiceToSet);
+      ensureEffect($effect`Transpondent`);
+      use($item`Map to Safety Shelter Grimace Prime`);
+      return true;
+    },
+  },
+  {
+    name: "Barf",
+    available: () => true,
+    perform: () => {
+      useFamiliar(barfFamiliar());
+      const lubing = get("dinseyRollercoasterNext") && have($item`lube-shoes`);
+      meatOutfit(
+        false,
+        lubing ? new Requirement([], { forceEquip: $items`lube-shoes` }) : undefined
```

Comment:
what is undefined here in the meatoutfit doing

Reply:
it's saying "if we aren't lubing, we aren't passing a Requirement here"

Reply:
we should have `meatOutfit` take an options object rather than the current structure probably, just for clarity

Reply:
I will probably start to handle that as I work on stateful outfit code. Which may or may not happen in the near future.